docs/docs/inference.rst

```diff
@@ -0,0 +1,114 @@
+.. _inference:
+
+==============
+Inference
+==============
+
+.. note::
+
+   This tutorial requires a :ref:`Unitxt installation <install_unitxt>`.
+
+Introduction
+------------
+Unitxt offers a wide array of :class:`Inference Engines <unitxt.inference>` for running models either locally (using HuggingFace, Ollama, and VLLM) or by making API requests to services like WatsonX, AWS, and Together AI.
+
+Unitxt inference engines serve two main purposes:
+
+    1. Running a full end-to-end evaluation pipeline with inference.
+    2. Using models for intermediate steps, such as evaluating other models (e.g., LLMs as judges) or for data augmentation.
+
+Running Models Locally
+-----------------------
+You can run models locally with inference engines like:
+
+    - :class:`HFPipelineBasedInferenceEngine <unitxt.inference.HFPipelineBasedInferenceEngine>`
+    - :class:`VLLMInferenceEngine <unitxt.inference.VLLMInferenceEngine>`
+    - :class:`OllamaInferenceEngine <unitxt.inference.OllamaInferenceEngine>`
+
+To get started, prepare your engine:
+
+.. code-block:: python
+
+    engine = HFPipelineBasedInferenceEngine(
+        model_name="meta-llama/Llama-3.2-1B", max_new_tokens=32
+    )
+
+Then load the data:
+
+.. code-block:: python
+
+    dataset = load_dataset(
+        card="cards.xsum",
+        template="templates.summarization.abstractive.formal",
+        format="formats.chat_api",
+        metrics=[llm_judge_with_summary_metric],
+        loader_limit=5,
+        split="test",
+    )
+
+Notice: we create the data with  `format="formats.chat_api"` which produce data as list of chat turns:
+
+.. code-block:: python
+
+    [
+        {"role": "system", "content": "Summarize the following Document."},
+        {"role": "user", "content": "Document: <...>"}
+    ]
+
+Now run inference on the dataset:
+
+.. code-block:: python
+
+    predictions = engine.infer(dataset)
+
+Finally, evaluate the predictions and obtain final scores:
+
+.. code-block:: python
+
+    evaluate(predictions=predictions, data=dataset)
+
+Calling Models Through APIs
+---------------------------
+Calling models through an API is even simpler and is primarily done using one class: :class:`CrossProviderInferenceEngine <unitxt.inference.CrossProviderInferenceEngine>`.
```

Comment:
We should add at some point the HF  as a provider. So I would not split it to between APIs and local.  Instead you can call specific inference engine , or use the CrossProvider inference engine, which gives a standard API to multiple providers.